struct_darwin_arm64.go

```diff
@@ -0,0 +1,252 @@
+		}
+	} else if size <= 16 {
+		// Non-HFA/HVA small structs use int registers for byte-packing
+		slotsNeeded := int((size + align8ByteMask) / align8ByteSize)
+		if tempNumInts+slotsNeeded <= numOfIntegerRegisters() {
+			return true, tempNumInts + slotsNeeded, tempNumFloats
+		}
+	}
+
+	return false, tempNumInts, tempNumFloats
+}
+
+// collectStackArgs separates remaining arguments into those that fit in registers vs those that go on stack.
+// It returns the stack arguments and processes register arguments through addValue.
+func collectStackArgs(args []reflect.Value, startIdx int, numInts, numFloats int,
+	keepAlive []any, addInt, addFloat, addStack func(uintptr),
+	pNumInts, pNumFloats, pNumStack *int) ([]reflect.Value, []any) {
+
+	var stackArgs []reflect.Value
+	tempNumInts := numInts
+	tempNumFloats := numFloats
+
+	for j, val := range args[startIdx:] {
+		// Determine if this argument goes to register or stack
+		var fitsInRegister bool
+		var newNumInts, newNumFloats int
+
+		if val.Kind() == reflect.Struct {
+			// Check if struct still fits in remaining registers
+			fitsInRegister, newNumInts, newNumFloats = structFitsInRegisters(val, tempNumInts, tempNumFloats)
+		} else {
+			// Primitive argument
+			isFloat := val.Kind() == reflect.Float32 || val.Kind() == reflect.Float64
+			if isFloat {
+				fitsInRegister = tempNumFloats < numOfFloatRegisters
+				newNumFloats = tempNumFloats + 1
+				newNumInts = tempNumInts
+			} else {
+				fitsInRegister = tempNumInts < numOfIntegerRegisters()
+				newNumInts = tempNumInts + 1
+				newNumFloats = tempNumFloats
+			}
+		}
+
+		if fitsInRegister {
+			// Process through normal register allocation
+			tempNumInts = newNumInts
+			tempNumFloats = newNumFloats
+			keepAlive = addValue(val, keepAlive, addInt, addFloat, addStack, pNumInts, pNumFloats, pNumStack)
+		} else {
+			// Convert strings to C strings before bundling
+			if val.Kind() == reflect.String {
+				ptr := strings.CString(val.String())
+				keepAlive = append(keepAlive, ptr)
+				val = reflect.ValueOf(ptr)
+				args[startIdx+j] = val
+			}
+			stackArgs = append(stackArgs, val)
+		}
+	}
+
+	return stackArgs, keepAlive
+}
+
+const (
+	paddingFieldPrefix = "Pad"
+)
+
+// bundleStackArgs bundles remaining arguments for Darwin ARM64 C-style stack packing.
+// It creates a packed struct with proper alignment and copies it to the stack in 8-byte chunks.
+func bundleStackArgs(stackArgs []reflect.Value, addStack func(uintptr)) {
+	if len(stackArgs) == 0 {
+		return
+	}
+
+	// Build struct fields with proper C alignment and padding
+	var fields []reflect.StructField
+	currentOffset := uintptr(0)
+	fieldIndex := 0
+
+	for j, val := range stackArgs {
+		valSize := val.Type().Size()
+		valAlign := val.Type().Align()
+
+		// ARM64 requires 8-byte alignment for 8-byte or larger structs
+		if val.Kind() == reflect.Struct && valSize >= 8 {
+			valAlign = 8
+		}
+
+		// Add padding field if needed for alignment
```

Comment:
If we are now aligning struct arguments then we should update the [docs](https://pkg.go.dev/github.com/ebitengine/purego#hdr-Structs-RegisterFunc). I'm not sure if we want to be responsible for it though.